packages/trpc/server/routers/viewer.tsx

```diff
@@ -392,6 +425,11 @@ const loggedInViewerRouter = createProtectedRouter()
                 },
               },
             },
+            {
+              eventTypeId: {
+                in: eventTypes.map((event) => event.id),
+              },
+            },
```

Comment:
We can solve this in one query instead of four:

```suggestion
            {
              eventType: {
                team: {
                  members: {
                    some: {
                      userId: user.id,
                      role: "OWNER",
                    },
                  },
                },
              },
            },
```